src/library/coverartcache.cpp

```diff
@@ -146,9 +147,13 @@ CoverArtCache::FutureResult CoverArtCache::loadCover(
 
 // watcher
 void CoverArtCache::coverLoaded() {
-    QFutureWatcher<FutureResult>* watcher;
-    watcher = reinterpret_cast<QFutureWatcher<FutureResult>*>(sender());
-    FutureResult res = watcher->result();
+    FutureResult res;
+    {
```

Comment:
What does the scope here?

Reply:
To ensure that the watcher is not used anywhere else.